src/Type/UnionType.php

```diff
@@ -137,6 +139,13 @@ public function isSuperTypeOf(Type $otherType): TrinaryLogic
 			return $otherType->isSubTypeOf($this);
 		}
 
+		if ($otherType instanceof IntegerRangeType && $otherType->getMin() !== null && $otherType->getMax() !== null) {
```

Comment:
Would be much nicer if this logic was in ConstantIntegerType or IntegerRangeType :) If we need to make IntegerRangeType implement CompoundType (so that it has isSubTypeOf/isAcceptedBy methods) I'm fine with it (hopefully it doesn't break anything)